python/vast/utils/arrow.py

```diff
@@ -11,3 +150,21 @@ def names(schema: pa.Schema):
 def name(schema: pa.Schema):
     xs = names(schema)
     return xs[0] if xs[0] else ""
+
+
+# Accepts a 128-bit buffer holding an IPv6 address and returns an IPv4 or IPv6
+# address.
+def unpack_ip(buffer: SupportsBytes) -> ip.IPv4Address | ip.IPv6Address:
+    num = int.from_bytes(buffer, byteorder="big")
+    # Convert IPv4 mapped addresses back to regular IPv4.
+    # https://tools.ietf.org/html/rfc4291#section-2.5.5.2
+    if (num >> 32) == 65535:
+        num = num - (65535 << 32)
+    return ip.ip_address(num)
+
+
+# Modules are intialized exactly once, so we can perform the registration here.
+pa.register_extension_type(PatternType())
+pa.register_extension_type(AddressType())
+pa.register_extension_type(SubnetType())
+pa.register_extension_type(EnumType({"stub": 0}))
```

Comment:
@dominiklohmann suggested this because we're doing this in C++.

> Isn't registration required for each specific enum definition anyway?

I'm actually skeptical that our approach works at all, because registration of an extension type happens by name. We currently register this particular instance under the type name `vast.enumeration`. Registering *another* enum fails.

Reply:
What about adding the hash of the `_fields` to the name

Reply:
This is required to mirror what we do on the C++ side of things for compatibility.

Reply:
But how does it even work in C++ land?

Reply:
The registration only ever calls member functions of the extension type that are required to have a constant output regardless of the types instantiation. Essentially, you don't register an instance of the extension type, but rather the extension type itself. Registering only takes an instance of the type because it's a badly designed API.